### PR TITLE
:window: :wrench: Remove demo mode from UI

### LINKED ISSUE #15648: Remove demo mode from UI
We currently have the possibility to specify a demo mode (if `window.IS_DEMO` is set to true). This mode is only used to show a banner notification that you're in the demo. We should use any of our third party tools (like google optimize) to show a banner instead.

Long term the demo instance should move to Cloud: https://github.com/airbytehq/airbyte/issues/14765

* Remove demo mode from webapp
* Remove `IS_DEMO` from build configurations and docs.
* Remove any other areas not mentioned here (search for `isDemo` and `IS_DEMO` in code)
* Verify that there are no `IS_DEMO` references in `airbyte-cloud`. Do not remove helm charts
* Coordinate with @nataliekwong to add banner notification - Deploy built version and show banner with third-party tool

### PR DESCRIPTION
**Pull request by lmossman**
## What
Resolves https://github.com/airbytehq/airbyte/issues/15648

## How
Removes the demo mode banner from the UI, as well as any code related to that banner (e.g. env vars and intl messages), as we will now be using a third-party tool to display that banner on the demo airbyte deployment.

There is a separate PR to remove the env var from the one place it exists in the cloud repo as well: https://github.com/airbytehq/airbyte-cloud/pull/2397

### CODE REVIEW

**Review by krishnaglick — COMMENTED** [review]
From a code-overview it looks fine. I'm not sure how I could test this locally.

**Comment by lmossman** [comment]
Yeah, I'm not really sure how to test this locally either, since this is just removing all of the IS_DEMO stuff, which is only really relevant to the demo deployment. 

So it feels like all we can really do is wait until this is merged, and then deploy the demo application to see if it looks correct (which I expect it to since it should now be the same as non-demo).

I suppose we could also deploy the demo app from this branch so we can test it out before merging, but I'm not exactly sure how that is done. @nataliekwong do you know how we currently deploy the demo instance, or do you know who would know about that?

**Comment by nataliekwong** [comment]
Yes! @marcosmarxm directed me to this page before on how to access it: https://www.notion.so/Demo-Airbyte-Instance-b2b8f1ddfa53438db78433cca0ae0568

**Comment by lmossman** [comment]
Thank you for linking that, super helpful!

@krishnaglick in that Notion page linked above, it says that the command to deploy airbyte in the demo instance is
```
API_URL=/api/v1/ AIRBYTE_ROLE=demo IS_DEMO=true docker-compose up -d
```
So I tried running that locally on this PR branch, and I confirmed that the demo mode banner is no longer present in the UI, and the rest of the app seems to be working as expected. So I think that should be sufficient for local testing

**Comment by marcosmarxm** [comment]
@lmossman this won't break the demo.airbyte.io stuff?

File: charts/airbyte-webapp/values.yaml
Comment (edmundito) [inline charts/airbyte-webapp/values.yaml:180]: Unresolved conflict, whoops

**Review by edmundito — CHANGES_REQUESTED** [review]
Overall it's looking good. I found an issue that needs to be fixed.

**Comment by edmundito** [comment]
@marcosmarxm See https://github.com/airbytehq/airbyte/issues/15648 for details. This should not be merged until the new way to set the messaging is set up and ready to be enabled.

**Review by krishnaglick — APPROVED** [review]
I'm marking this as approved, pending Edmundo's greenlight and that merge message removed 👍

File: charts/airbyte-webapp/values.yaml
Comment (lmossman) [reply charts/airbyte-webapp/values.yaml]: Good catch! Looks like I used the vs code "accept incoming change" button incorrectly somehow. Pushed the fix!

**Comment by lmossman** [comment]
Aha, I found where we configured the demo app to fail non-read API calls: https://github.com/airbytehq/airbyte/blob/381326f4d32760119697768857c669026911dc30/terraform/aws/demo/lb/main.tf#L87-L107
Since that is configured through terraform, that should not be affected by any of the changes in this PR. So this should be safe to merge!

**Pull request merged**
